runtime/src/bank.rs

```diff
@@ -5270,6 +5271,24 @@ impl Bank {
         );
         load_time.stop();
 
+        let preexecution_account_states: Option<HashMap<Pubkey, AccountSharedData>> = self
```

Comment:
We should make clear to operators that enabling this (and thus including this statement), will quite significantly slow down runtime due to all of the allocations here. This runs even for transactions geyser doesn't care about.

Reply:
Will add a note in geyser plugin interface that it can slow the runtime.

Tracking from geyser is impossible and does not make a lot of sense, we have to track all the accounts all the time because account filters can be changed by subscription. Synchronizing old account states with the new ones would also add new constraints. Adding this additional complexity to the geyser plugin can cause more severe performance degradation. Bootstraping the old accounts would also be an issue. In this implementation, accounts are already loaded for transaction execution, and just copying data from one part of memory to another will cause way less performance impact. As it is opt-in and only enabled if needed by the geyser plugin, these plugins would only be enabled on an rpc server. So this is the best possible solution.

Reply:
If you are tracking all the accounts by owner, you will miss these account deleted notifications, because there may not be any prior notifications for that account. Again these allocations are opt in and won't affect any voting validators. You have a choice not to use this feature and implement whatever strategy you want in your geyser plugin implementation.

Reply:
You can pre-populate your geyser plugin with the needs keys if this odd edge case is that important. 

Regardless, I'm against this PR -- in favor instead of a much larger refactor of the bank. This way we don't add even more bloat/branches to code that already needs simplification and refactoring.